### ISSUE TITLE
Flatpak clients will not run via Game Mode on Steam Deck

### ISSUE THREAD

**Issue by ImDink** [TormStorm/jagex-launcher-linux#83]
**Describe the bug**
The HDOS client will not launch via Game Mode on the Steam Deck via the Jagex Launcher, however it is able to open via Desktop Mode.

**To Reproduce**
1. Installed Jagex Launcher via Lutris local script
2. Logged into the launcher via Desktop mode and confirmed is working
3. Swapped back to Game Mode
4. Attempted to launch HDOS, play button changes to '...' momentarily then nothing changes and play button is clickable again
(Runelite is able to open without issue)

**Expected behavior**
HDOS client launches and I can enjoy some afk gameplay while in the office or on a train :)

**Screenshots**
None taken, but I can if needed

**Please complete the following information:**
 - Steam OS 3.5
 - KDE Plasma with desktop mode, not sure in game mode (steam?)

**Additional context**
I love what you've done, and I'm learning a lot via seeing how these things work so I thank you <3

**Comment by TormStorm** [comment]
Hello, I updated the HDOS script earlier today, can you try to update it and see if it helps?
https://github.com/TormStorm/jagex-launcher-linux/blob/main/resources/hdos.sh

**Comment by nmlynch94** [comment]
That change shouldn't make a difference since he's using flatpak, esp since it works on desktop mode. Usually something like this is due to the settings for the client not playing nicely with the Steam Decks gamescope session. I can mess around with some settings.

**Comment by nissan24zero** [comment]
Following.

**Comment by ImDink** [comment]
As nmlynch94 suggested, there was no change in game mode.

I'm not sure how much this information helps, but when I launch the HDOS client itself via game mode without the Jagex launcher it loads up just like RuneLite would with the splash window and a progress bar, and then it opens in a cropped window for the client itself, also using the flatpak

**Comment by nmlynch94** [comment]
Yeah I got it to work, but I had to install the windows version into the prefix instead of using the flatpak HDOS client, which seems to be completely broken for game mode. I also could only get it to launch consistently using bottles and not Lutris. I installed the jagex launcher and then hdos win executables into the bottle. Trying to full screen or opening the wrong menu made it overflow the game mode screen, so I couldn't touch many settings safely besides plugin settings lol. It did work though even though it's janky as long as you just play and don't try to adjust settings in game mode.

The unfortunately reality is that the client just does not behave will in game mode as it currently stands. It doesn't allow you to set the resolution and lock the window size like in runelite so the client doesn't escape the boundary. Desktop mode is your best bet for now I think.

That being said, feel free to try it if you'd like. Here is my bottle config: https://pastebin.com/U4tyj0jz. You just need to install the jagex launcher and then hdos windows versions into it and add a shortcut to steam. If jagex launcher times out, try toggling discrete graphics off and running again.

**Comment by Otkar** [comment]
For what it's worth, I was able to revert back to using RuneLite via the `.AppImage` format. Could be a temporary method until this get fixed.

**Comment by nmlynch94** [comment]
Further context, I believe this is the root cause:

```
java.awt.HeadlessException: 
No X11 DISPLAY variable was set, but this program performed an operation which requires it.
    at java.desktop/java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:208)
    at java.desktop/java.awt.Window.<init>(Window.java:548)
    at java.desktop/java.awt.Frame.<init>(Frame.java:423)
    at java.desktop/java.awt.Frame.<init>(Frame.java:388)
    at java.desktop/javax.swing.JFrame.<init>(JFrame.java:180)
    at net.runelite.launcher.SplashScreen.<init>(SplashScreen.java:71)
    at net.runelite.launcher.SplashScreen.lambda$init$0(SplashScreen.java:175)
    at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:303)
    at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
    at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
    at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
    at java.base/java.security.AccessController.doPrivileged(Native Method)
    at 
...

I tried manually setting the DISPLAY env var in flatpak-spawn, but no luck so far.

**Comment by TormStorm** [comment]
Resolved with #100
